dask/dataframe/core.py

```diff
@@ -3748,7 +3749,9 @@ def _validate_axis(cls, axis=0):
     @derived_from(pd.DataFrame)
     def drop(self, labels=None, axis=0, columns=None, errors="raise"):
         axis = self._validate_axis(axis)
-        if (axis == 1) or (columns is not None):
+        if columns is not None:
+            return self.map_partitions(drop_by_shallow_copy, columns)
```

Comment:
Previously for `axis==0` we would raise.  I think that we're still supposed to do that here.

Reply:
It also looks like we're ignooring the labels and  errors keywords.  I honestly don't know much about these to know if they're important or not though.

Reply:
You specify either `labels` and `axis` or `index` and `columns` (and Dask doesn't have the `index` keyword). So we'll want to restore that original check.